kythe/cxx/indexer/cxx/marked_source.cc

```diff
@@ -588,9 +588,24 @@ std::string GetDeclName(const clang::LangOptions& lang_options,
   return "";
 }
 
+namespace {
+class : public clang::DiagnosticConsumer {
+  void HandleDiagnostic(clang::DiagnosticsEngine::Level,
+                        const clang::Diagnostic&) override {}
+} ignoring_diagnostic_consumer;
```

Comment:
just use `clang::IgnoringDiagConsumer`

Reply:
(And you should just construct a new one each time, rather than using a global.  They aren't entirely stateless but are very cheap to construct).